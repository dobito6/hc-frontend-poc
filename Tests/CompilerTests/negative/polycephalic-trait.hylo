//! no-std

trait P<T> {
  fun f() -> T
}

struct A {
  public memberwise init
}

fun check<T>(_ x: T) {}

fun g<X, Y where X is P<Y>>(_ x: X) {
  check<Y>(x.f())
}

given A is P<Void> {
  public fun f() -> Void { () }
}

public fun main() {
  g<A, Void>(A())
  g<A, A>(A())
  g<A, _>(A())
}
