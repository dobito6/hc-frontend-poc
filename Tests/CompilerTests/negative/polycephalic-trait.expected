negative/polycephalic-trait.hylo:23.3-10: error: no coercion from 'P<A, A> ==> [Void](let A) let -> Void' to '[Void](let A) let -> Void'
  g<A, A>(A())
  ~~~~~~~
negative/polycephalic-trait.hylo:23.3-10: note: no given instance of 'P<A, A>' in this scope
  g<A, A>(A())
  ~~~~~~~
